### PR TITLE
test(client): Rewrite `tracing` test to not use snapshots

### PR DESCRIPTION
**Pull request by SevInf**
Since traces are delivered asynchronously, we can't reliably snapshot them - relevant traces might've not been delivered yet and snapshot might be flaky. To attempt to alleviate that we had 500ms delay in each tracing test which unnecessary increased test runtime in some cases (70 sequential tests, 0.5s each = 35 extra seconds) and still was not enough in others. Moving them to `toEqual` instead of snapshots allowed us to wrap assertions into `waitFor` and make them more stable, while also reducing the runtime.

Unfortinately, that did not fix main source of flakiness - itx test still sometimes misses tracing spans when run on data proxy. Debugging shows that the relevant spans are never returned from the engine side in this case - so it's a real bug, not a test problem.

### CODE REVIEW

**Review by Jolg42 — COMMENTED** [review]
I think you could rename this PR to either 
`test(client):`
or 
`fix(client)`

Feat would be for something new I think

File: packages/client/src/runtime/getPrismaClient.ts [hunk 1]
```diff
@@ -850,7 +850,7 @@ Or read our docs at https://www.prisma.io/docs/concepts/components/prisma-client
           attributes: {
             method: params.action,
             model: params.model,
-            name: `${params.model}.${params.action}`,
```
Comment (Jolg42) [inline packages/client/src/runtime/getPrismaClient.ts:853]: So for example, `$queryRaw`?

File: packages/client/tests/functional/tracing/tests.ts [hunk 2]
```diff
@@ -510,72 +334,46 @@ testMatrix.setupTestSuite(({ provider }, suiteMeta, clientMeta) => {
...
+              { name: 'prisma:engine:itx_query_builder', children: [findManyDbQuery(), engineSerialize()] },
+            ],
+          },
+        ],
+      })
     })
   })
 
-  describe('tracing on $raw methods', () => {
+  describeIf(provider !== 'mongodb')('tracing on $raw methods', () => {
```
Comment (Jolg42) [inline packages/client/tests/functional/tracing/tests.ts:367]: It looks like we're missing a test for https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#runcommandraw

Comment (Jolg42) [reply packages/client/tests/functional/tracing/tests.ts:367]: Can probably be done separately

**Review by Jolg42 — APPROVED** [review]
So much better than before 😍 

Thank you 🙌🏼

File: packages/client/src/runtime/getPrismaClient.ts (hunk 1 above)
Comment (SevInf) [reply packages/client/src/runtime/getPrismaClient.ts:853]: Oh,  right, completely forgot that I did this part. 
Yes, this is a fix for raw methods.

**Pull request merged**